### ISSUE TITLE
Getting started docs do not work in minikube

### ISSUE THREAD

**Issue by dictvm** [linkerd/linkerd2#1302]
I'm trying to get started with Linkerd2 in order to check out if we could use this in our clusters. However, I can't get it to work in minikube. 

I've followed the [docs](https://conduit.io/getting-started/).

`minikube version: v0.23.0`

`minikube start --kubernetes-version v1.10.0`
I'm on `macOS 10.13.5 (17F77)`. 

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.0", GitCommit:"91e7b4fd31fcd3d5f436da26c980becec37ceefe", GitTreeState:"clean", BuildDate:"2018-06-27T22:29:25Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-04-10T12:46:31Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"linux/amd64"}
```

```
$ conduit check
kubernetes-api: can initialize the client..................................[ok]
kubernetes-api: can query the Kubernetes API...............................[ok]
kubernetes-api: is running the minimum Kubernetes API version..............[ok]
conduit-api: can query the Conduit API.....................................[ERROR] -- error reading byte stream header: error while reading bytes from message: unexpected EOF
conduit-version: control plane is up-to-date...............................[ERROR] -- error reading byte stream header: error while reading bytes from message: unexpected EOF

Status check results are [ERROR]
```

```
$ kubectl -n conduit logs $(kubectl -n conduit get po | grep web | cut -f1 -d' ') conduit-proxy
INFO conduit_proxy using controller at Some(HostAndPort { host: DnsName(DnsName(DNSName("proxy-api.conduit.svc.cluster.local"))), port: 8086 })
INFO conduit_proxy routing on V4(127.0.0.1:4140)
INFO conduit_proxy proxying on V4(0.0.0.0:4143) to None
INFO conduit_proxy serving Prometheus metrics on V4(0.0.0.0:4191)
INFO conduit_proxy protocol detection disabled for inbound ports {25, 3306}
INFO conduit_proxy protocol detection disabled for outbound ports {25, 3306}
ERR! admin={bg=resolver} conduit_proxy::control::util controller error: Error attempting to establish underlying session layer: operation timed out after 3s
ERR! trust_dns_proto::xfer::dns_future error in Proto: proto error
WARN trust_dns_proto::xfer error notifying wait, possible future leak: Err(ResolveError { inner: ProtoError { inner:

request timed out }

request timed out })
ERR! admin={bg=resolver} conduit_proxy::control::util controller error: Error attempting to establish underlying session layer: operation timed out after 3s
WARN trust_dns_proto::xfer error notifying wait, possible future leak: Err(ResolveError { inner: ProtoError { inner:
```

```
$ kubectl logs controller-6557864bd7-xcw6j --namespace conduit public-api
time="2018-07-11T09:40:58Z" level=info msg="running conduit version v0.5.0"
time="2018-07-11T09:40:58Z" level=info msg="starting admin server on :9995"
time="2018-07-11T09:40:58Z" level=info msg="waiting for caches to sync"
time="2018-07-11T09:40:58Z" level=info msg="starting HTTP server on :8085"
E0711 09:40:58.682285       1 reflector.go:205] k8s.io/client-go/informers/factory.go:87: Failed to list *v1.ReplicationController: Get https://10.96.0.1:443/api/v1/replicationcontrollers?limit=500&resourceVersion=0: x509: certificate is valid for 192.168.99.100, 10.0.0.1, not 10.96.0.1
E0711 09:40:58.683107       1 reflector.go:205] k8s.io/client-go/informers/factory.go:87: Failed to list *v1.Service: Get https://10.96.0.1:443/api/v1/services?limit=500&resourceVersion=0: x509: certificate is valid for 192.168.99.100, 10.0.0.1, not 10.96.0.1
E0711 09:40:58.683347       1 reflector.go:205] k8s.io/client-go/informers/factory.go:87: Failed to list *v1beta2.Deployment: Get https://10.96.0.1:443/apis/apps/v1beta2/deployments?limit=500&resourceVersion=0: x509: certificate is valid for 192.168.99.100, 10.0.0.1, not 10.96.0.1
E0711 09:40:58.690135       1 reflector.go:205] k8s.io/client-go/informers/factory.go:87: Failed to list *v1.Pod: Get https://10.96.0.1:443/api/v1/pods?limit=500&resourceVersion=0: x509: certificate is valid for 192.168.99.100, 10.0.0.1, not 10.96.0.1
E0711 09:40:58.691661       1 reflector.go:205] k8s.io/client-go/informers/factory.go:87: Failed to list *v1beta2.ReplicaSet: Get https://10.96.0.1:443/apis/apps/v1beta2/replicasets?limit=500&resourceVersion=0: x509: certificate is valid for 192.168.99.100, 10.0.0.1, not 10.96.0.1
E0711 09:40:58.703484       1 reflector.go:205] k8s.io/client-go/informers/factory.go:87: Failed to list *v1.Namespace: Get https://10.96.0.1:443/api/v1/namespaces?limit=500&resourceVersion=0: x509: certificate is valid for 192.168.99.100, 10.0.0.1, not 10.96.0.1
```

Let me know if I should get something else from the logs.

**Comment by klingerf** [comment]
I was able to reproduce this with minikube v0.23.0, kubernetes v1.10.0, and conduit v0.5.0.

If I upgrade minikube to v0.28.0 (the latest), the problem goes away.

My guess is that minikube v0.23.0 can't reliably install kubernetes v1.10.0, since it was released ~6 months before 1.10 availability. Can you try upgrading your version of minikube to see if that fixes it?

**Comment by dictvm** [comment]
Damn, I ran `brew update && brew upgrade` but I didn't consider that minikube was installed as a cask and cask packages aren't upgraded that way. After upgrading minikube, Conduit is behaving fine.
